state/interface.go

```diff
@@ -25,6 +25,14 @@ type State interface {
 	Get(*cid.Cid) api.Pin
 	// Snapshot writes a snapshot of the state to a writer
 	Snapshot(w io.Writer) error
-	// Restore restores a snapshot from a reader
-	Restore(r io.Reader) error
+	// Restore restores an outdated state to the current version
+	Restore() error
```

Comment:
If Restore() is called from main (see comment above), it would be called on the mapstate directly and maybe doesn't need to be part of the interface (same as Snapshot). I think also GetLatestVersion() might be ok to remove too.

Reply:
Hmm... fair enough, I am wondering though what kind of state would not want a restore method?  To me it makes sense to keep that one on the interface even if we don't need to.  I will be happy to remove GetLatestVersion() now that I am bringing more things to main.

Reply:
Well, the interface is more, 'what are the requirements of the state components so it can be used by cluster'. If cluster does not use the Restore method... but yeah, it doesn't bother me much anyway.